src/cluster.c

```diff
@@ -1483,11 +1483,127 @@ static void addClusterSlotsReply(RedisRaftCtx *rr, RedisModuleCtx *ctx)
     RedisModule_ReplySetArrayLength(ctx, num_slots);
 }
 
+static void addClusterShardsNodeReply(RedisModuleCtx *ctx, char *id, uint16_t port, char *host, char *role, int offset, char *health)
+{
+    RedisModule_ReplyWithMap(ctx, 6);
+    RedisModule_ReplyWithCString(ctx, "id");
+    RedisModule_ReplyWithCString(ctx, id);
+    RedisModule_ReplyWithCString(ctx, "port");
+    RedisModule_ReplyWithLongLong(ctx, port);
+    RedisModule_ReplyWithCString(ctx, "ip");
+    RedisModule_ReplyWithCString(ctx, host);
+    RedisModule_ReplyWithCString(ctx, "role");
+    RedisModule_ReplyWithCString(ctx, role);
+    RedisModule_ReplyWithCString(ctx, "replication-offset");
+    RedisModule_ReplyWithLongLong(ctx, offset);
+    RedisModule_ReplyWithCString(ctx, "health");
+    RedisModule_ReplyWithCString(ctx, health);
```

Comment:
I think we need to verify if clients are handling this response correctly. Maybe we should try py-redis? 
I see some issues might be problematic, just listing here:

1- `endpoint` field is missing. It isn't optional according to doc. Here, we should return our hostname and port if I understand correctly.

2- Currently, we do not have support for NAT. Apparently, `endpoint` is also used for this purpose according to doc. (Maybe, we can ignore it for now )

3- We are returning hostname for the `ip` field. Not sure clients will be happy with this. (Note: I also don't understand the reasoning behind `ip` field. Hope I'm missing something, otherwise it doesn't look like a good API. Should this be an IP? Why can't this be a hostname?). 

4- Will TLS work with current clients? We are not setting `tls-port` field. Will clients open tls connection even when they don't see `tls-port` field?

Reply:
Hmm, this is introduced in 7.0. I guess we have to implement this blindy? Without trying with a client.

Reply:
good catch, I did it based on documented examples, but I see they list those additional fields in the text portion of the documentation.

https://redis.io/commands/cluster-shards/

Reply:
1- Regarding `endpoint`, problem is in the PR and docs.

For `cluster shards`: Redis adds `endpoint` field to the response.
For `cluster slots`: Endpoint is either `hostname` or `ip` field. There is no separate `endpoint` field.

I'm not sure if this is documented well or the intended behavior. So, I guess we should either fix the code or the documentation after deciding the correct behavior. 

--------
2-
> ip -> do a resolution on hostname

I'm having hard time to understand this field. Are we trying to help the client so it wouldn't need to resolve a hostname? Why should we return this when our cluster is configured with hostnames?. We need to resolve all hostnames in the cluster (including other shard groups)? I think we should decide if it is okay to return empty string if this field is mandatory.  Then, we can just return hostname in the `endpoint` field. Client can resolve the hostname itself.

Reply:
* Endpoint - yes, it was added to `cluster slot` in Redis7.0 - we should update this command too (will open a separate ticket for that). Anyway, this field should depend on `cluster-preferred-endpoint-type` configuration in both cases.
* I guess we can stick to the Redis code and suggest fixing to the documentation.Please add Harkrishn Patro (who added `cluster shards`) and Madelyn Olson (`cluster slots` update) to this discussion to ensure the documentation should be updated.
*  Regarding `ip` - why not to keep the ip too? any reason why `NodeAddr.host` can be hostname or ip?